pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -178,7 +178,11 @@ func (r *hibernationReconciler) Reconcile(ctx context.Context, request reconcile
 
 	// If cluster is already deleted, skip any processing
 	if !cd.DeletionTimestamp.IsZero() {
-		return reconcile.Result{}, nil
+		cd.Status.PowerState = hivev1.ClusterPowerStateUnknown
+		msg := "ClusterDeployment has been marked for deletion"
+		r.setCDCondition(cd, hivev1.ClusterHibernatingCondition, hivev1.HibernatingReasonClusterDeploymentDeleted, msg, corev1.ConditionUnknown, cdLog)
+		r.setCDCondition(cd, hivev1.ClusterReadyCondition, hivev1.ReadyReasonClusterDeploymentDeleted, msg, corev1.ConditionUnknown, cdLog)
+		return reconcile.Result{}, r.updateClusterDeploymentStatus(cd, cdLog)
```

Comment:
We need to be careful not to issue the Update unless we've actually changed something. No-op updates can end up thrashing with other controllers. `setCDCondition` returns a bool indicating whether it changed anything; and you can manually check the status.powerState. Perform `updateClusterDeploymentStatus` iff any of those things are true, but not if they're all false.

Reply:
@2uasimojo Do you think it would be fair to pull this section out into a separate function, as the behavior is near identical to https://github.com/openshift/hive/blob/master/pkg/controller/hibernation/hibernation_controller.go#L202:L216

Reply:
Yup, I like.